lib/deride.js

```diff
@@ -284,8 +284,13 @@ function Setup(obj, method, emitter) {
         if (_.isFunction(callBasedOnArgs)) {
             return callBasedOnArgs.apply(self, arguments);
         }
-        if (_.isFunction(callBasedOnPredicate)) {
-            return callBasedOnPredicate.apply(self, arguments);
+
+        var args = arguments;
+        var matchingPredicateTuple = functionPredicates.find(function(tuple){
+            return tuple[0].apply(self, args);
+        });
+        if (matchingPredicateTuple) {
+            return matchingPredicateTuple[1].apply(self, arguments);
```

Comment:
Took me a few minutes to understand why you were indexing with a magic number of 1 here until I noticed line 395 where you are using an array as a tuple.  Never done that before in javascript but I like the idea - nice work.